pkg/reconciler/taskrun/taskrun.go

```diff
@@ -203,26 +194,38 @@ func (c *Reconciler) ReconcileKind(ctx context.Context, tr *v1beta1.TaskRun) pkg
 	}
 	return nil
 }
-func (c *Reconciler) stopSidecars(ctx context.Context, tr *v1beta1.TaskRun) (*corev1.Pod, error) {
+func (c *Reconciler) stopSidecars(ctx context.Context, tr *v1beta1.TaskRun) error {
 	logger := logging.FromContext(ctx)
 	// do not continue without knowing the associated pod
 	if tr.Status.PodName == "" {
-		return nil, nil
+		return nil
 	}
 
 	// do not continue if the TaskSpec had no sidecars
 	if tr.Status.TaskSpec != nil && len(tr.Status.TaskSpec.Sidecars) == 0 {
-		return nil, nil
+		return nil
 	}
 
 	// do not continue if the TaskRun was canceled or timed out as this caused the pod to be deleted in failTaskRun
 	condition := tr.Status.GetCondition(apis.ConditionSucceeded)
 	if condition != nil {
 		reason := v1beta1.TaskRunReason(condition.Reason)
 		if reason == v1beta1.TaskRunReasonCancelled || reason == v1beta1.TaskRunReasonTimedOut {
-			return nil, nil
+			return nil
+		}
+	}
+
+	// do not continue if there are no Running sidecars.
+	needsStop := false
+	for _, ss := range tr.Status.Sidecars {
+		if ss.Running != nil {
```

Comment:
What is difference between ss.Running v/s ss.Terminated? There is a function `IsSidecarStatusRunning` defined in package `pod`, can we use the same function here instead?

https://github.com/tektoncd/pipeline/blob/2ee13837eaee98ec19b5ca5e53e19d9e532433b6/pkg/reconciler/taskrun/taskrun.go#L231

Reply:
Excellent suggestion! Much nicer, thanks.